src/test/java/uk/gov/register/core/PostgresRegisterTest.java

```diff
@@ -41,4 +45,60 @@ public void findMax100RecordsByKeyValueShouldReturnValueWhenKeyExists() {
         register.max100RecordsFacetedByKeyValue("name", "United Kingdom");
         verify(backingStoreDriver, times(1)).findMax100RecordsByKeyValue("name", "United Kingdom");
     }
-}
\ No newline at end of file
+
+    @Test(expected = NoSuchItemForEntryException.class)
+    public void appendEntryShouldThrowExceptionIfNoCorrespondingItemExists() {
+        Entry entryDangling = new Entry(106, "item-hash-2", Instant.now());
+
+        when(backingStoreDriver.getItemBySha256(anyString())).thenReturn(Optional.empty());
+
+        PostgresRegister register = new PostgresRegister(registerData, backingStoreDriver, itemValidator);
+        register.appendEntry(entryDangling);
+    }
+
+    @Test
+    public void appendEntryShouldNotInsertDanglingEntry() {
+        Item item = new Item("item-hash-1", new ObjectMapper().createObjectNode());
+        Entry entryNotDangling = new Entry(105, "item-hash-1", Instant.now());
+        Entry entryDangling = new Entry(106, "item-hash-2", Instant.now());
+
+        when(backingStoreDriver.getItemBySha256(anyString()))
+                .thenReturn(Optional.of(item))
+                .thenReturn(Optional.empty());
+
+        PostgresRegister register = new PostgresRegister(registerData, backingStoreDriver, itemValidator);
+
+        try {
+            register.appendEntry(entryNotDangling);
+            register.appendEntry(entryDangling);
+        } catch (NoSuchItemForEntryException ex) {
+        }
+
+        verify(backingStoreDriver, times(1)).insertEntry(any());
+    }
+
+    @Test
+    public void appendEntryShouldNotInsertRecordForDanglingEntry() {
+        Item item = new Item("item-hash-1", new ObjectMapper().createObjectNode());
+        Entry entryNotDangling = new Entry(105, "item-hash-1", Instant.now());
+        Entry entryDangling = new Entry(106, "item-hash-2", Instant.now());
+
+        when(backingStoreDriver.getItemBySha256(anyString()))
+                .thenReturn(Optional.of(item))
+                .thenReturn(Optional.empty());
+
+        RegisterMetadata registerMetadata = mock(RegisterMetadata.class);
+        when(registerMetadata.getRegisterName()).thenReturn("country");
+        when(registerData.getRegister()).thenReturn(registerMetadata);
+
+        PostgresRegister register = new PostgresRegister(registerData, backingStoreDriver, itemValidator);
+
+        try {
+            register.appendEntry(entryNotDangling);
+            register.appendEntry(entryDangling);
+        } catch (NoSuchItemForEntryException ex) {
+        }
+
+        verify(backingStoreDriver, times(1)).insertRecord(any(), eq("country"));
```

Comment:
Similarly, we're not checking here the record that is inserted is the record we expect to be inserted.

```ArgumentCaptor<Record> record = ArgumentCaptor.forClass(Record.class);
        verify(backingStoreDriver, times(1)).insertRecord(record.capture(), eq("country"));
        assertThat(record.getValue().entry, equalTo(entryNotDangling));```